src/state/slices/limitOrderInputSlice/selectors.ts

```diff
@@ -63,14 +63,10 @@ export const selectExpiry = createSelector(selectBaseSlice, baseSlice => baseSli
 
 // This is the buy amount based on the quote + user input.
 export const selectBuyAmountCryptoBaseUnit = createSelector(
-  selectInputSellAmountCryptoPrecision,
-  selectInputBuyAsset,
+  selectInputSellAmountCryptoBaseUnit,
   selectLimitPrice,
-  (inputSellAmountCryptoPrecision, buyAsset, limitPrice) => {
-    // Arithmetic MUST be in CryptoPrecision due to differing decimals on various tokens.
-    return toBaseUnit(
-      bn(inputSellAmountCryptoPrecision).times(limitPrice.buyAssetDenomination),
-      buyAsset.precision,
-    )
+  (inputSellAmountCryptoBaseUnit, limitPrice) => {
+    // Arithmetic MUST be in CryptoBaseUnit to avoid precision loss converting back and forth.
+    return bn(inputSellAmountCryptoBaseUnit).times(limitPrice.buyAssetDenomination).toFixed()
```

Comment:
blocking: this won't work when dealing with multiple base units e.g:

<img width="1278" alt="image" src="https://github.com/user-attachments/assets/358995fd-3b20-48b0-a115-49c5b27bf6e8">

- 100123456781234567000 inputSellAmountCryptoBaseUnit (100.123456781234567000 FOX)
- 0.07238467339191986179 `limitPrice.buyAssetDenomination`

Yields `7247403717979668012.23008076421049493` which is off by sellAssetPrecision - buyAssetPrecision i.e 18 - 6 = 18

Reply:
Actually really well spotted. I've pushed a fix 3da61fbf5ed5eaa0058c36f4f2192304e559db1b to convert the precision to the buy asset precision here, and it looks to have resolved the issues noted here.

<img src="https://github.com/user-attachments/assets/c6db062b-e74b-45fd-85cf-c09c4871b64d" width="200">
<img src="https://github.com/user-attachments/assets/50861d4b-b24c-417d-be8c-592170c74e2d" width="200">